chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RealWorldTestingUploadJob.java

```diff
@@ -101,64 +127,71 @@ private RealWorldTestingUpload getRwtUploadFromFuture(CompletableFuture<RealWorl
-        request.setListing(listing);
+    private RealWorldTestingUpload updateListing(CertifiedProductSearchDetails listing, RealWorldTestingUpload rwt) {
         try {
+            rwt = addRwtDataToListing(listing, rwt);
+            ListingUpdateRequest request = new ListingUpdateRequest();
+            request.setAcknowledgeWarnings(true);
+            request.setListing(listing);
             certifiedProductManager.update(request);
         } catch (ValidationException e) {
-            rwtUpload.getErrors().addAll(e.getErrorMessages());
+            rwt.getValidationErrors().addAll(e.getErrorMessages());
         } catch (Exception e) {
-            rwtUpload.getErrors().add(e.getMessage());
+            rwt.getValidationErrors().add(e.getMessage());
         }
-        return rwtUpload;
+        return rwt;
     }
 
-    private boolean hasDataChanged(CertifiedProductSearchDetails listing, RealWorldTestingUpload rwtUpload) {
-        if (rwtUpload.getType().equals(RealWorldTestingType.PLANS)) {
-            return !(Objects.equal(listing.getRwtPlansUrl(), rwtUpload.getUrl())
-                    && Objects.equal(listing.getRwtPlansCheckDate(), rwtUpload.getLastChecked()));
-        } else if (rwtUpload.getType().equals(RealWorldTestingType.RESULTS)) {
-            return !(Objects.equal(listing.getRwtResultsUrl(), rwtUpload.getUrl())
-                    && Objects.equal(listing.getRwtResultsCheckDate(), rwtUpload.getLastChecked()));
-        } else {
-            return false;
+    private RealWorldTestingUpload addRwtDataToListing(CertifiedProductSearchDetails listing,
+            RealWorldTestingUpload rwt) {
+        if (rwt.getType().equals(RealWorldTestingType.PLANS)) {
+            listing.setRwtPlansCheckDate(rwt.getLastChecked());
+            listing.setRwtPlansUrl(rwt.getUrl());
+        } else if (rwt.getType().equals(RealWorldTestingType.RESULTS)) {
+            listing.setRwtResultsCheckDate(rwt.getLastChecked());
+            listing.setRwtResultsUrl(rwt.getUrl());
         }
+        return rwt;
+    }
+
+    private boolean hasDataRwtPlansChanged(CertifiedProductSearchDetails listing, RealWorldTestingUpload rwt) {
+        return !(Objects.equals(listing.getRwtPlansUrl(), rwt.getUrl())
+                && Objects.equals(listing.getRwtPlansCheckDate(), rwt.getLastChecked()));
+    }
+
+    private boolean hasDataRwtResultsChanged(CertifiedProductSearchDetails listing, RealWorldTestingUpload rwt) {
+        return !(Objects.equals(listing.getRwtResultsUrl(), rwt.getUrl())
+                && Objects.equals(listing.getRwtResultsCheckDate(), rwt.getLastChecked()));
     }
 
     @SuppressWarnings("checkstyle:linelength")
     private Optional<CertifiedProductSearchDetails> getListing(String chplProductNumber) {
         try {
-            return Optional.ofNullable(certifiedProductDetailsManager.getCertifiedProductDetailsByChplProductNumber(chplProductNumber));
+            return Optional.ofNullable(
```

Comment:
I think this should either be on one line, or the `@SuppressWarnings` of linelength should be removed